src/main/java/ch/njol/skript/lang/parser/ParserInstance.java

```diff
@@ -394,11 +399,68 @@ public Node getNode() {
 	public void setIndentation(String indentation) {
 		this.indentation = indentation;
 	}
-	
+
 	public String getIndentation() {
 		return indentation;
 	}
 
+	// Experiments API
+
+	@Override
+	public boolean hasExperiment(String featureName) {
+		return Skript.experiments().isUsing(this.getCurrentScript(), featureName);
+	}
+
+
+	@Override
+	public boolean hasExperiment(Experiment experiment) {
+		return Skript.experiments().isUsing(this.getCurrentScript(), experiment);
+	}
+
+	/**
+	 * Marks this as using an experimental feature.
+	 * @param experiment The feature to register.
+	 */
+	@ApiStatus.Internal
+	public void addExperiment(Experiment experiment) {
+		Script script = this.getCurrentScript();
+		@Nullable ExperimentSet set = script.getData(ExperimentSet.class);
+		if (set == null) {
+			set = new ExperimentSet();
+			script.addData(set);
+		}
+		set.add(experiment);
```

Comment:
```suggestion
		ExperimentSet set = script.getData(ExperimentSet.class, () -> new ExperimentSet());
		set.add(experiment);
```
you could provide a creation supplier instead